src/components/drawer/Drawer.js

```diff
@@ -0,0 +1,50 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useNavigation } from '../../hooks';
+
+const drawerClassName = 'react-navigation_drawer';
+
+const Drawer = ({ placement, withOverlay, children, className, overlayClassName }) => {
+    const { drawer } = useNavigation();
+
+    const drawerClassNames = classNames(drawerClassName, `${drawerClassName}-${placement}`, className);
```

Comment:
Because to use the data attribute I would need to do something like:
`.react-navigation_drawer[data-placement="top"]`

This creates a specificity of 20.

By having its own dedicated placement class, the specificity is still 10 and the className provided by prop can easily override it.